### PR TITLE
Windows32

### PR DESCRIPTION
**Pull request by ciechowoj**
Add CI for 32-bit.

### CODE REVIEW

**Comment by ciechowoj** [comment]
Tests for x86 works correctly.
Tests for x64 are failing (for some reason standard clang headers are not in the include search path).

**Comment by ciechowoj** [comment]
I can remove x64 from the matrix and IMO it could be ready to be merged. I'll fix problems with x64 in separate PR.

**Comment by ciechowoj** [comment]
For now I don't know what is going on. Tests for libcangs 3.4 - 3.5 are passing, and for more recent versions are failing due to missing standard header files : ( .

**Comment by jacob-carlborg** [comment]
Hmm, on Posix I assume that the user have the SDK installed already. Is it just a matter of setting up some paths so Clang can find them?

**Comment by ciechowoj** [comment]
All green : )

> Hmm, on Posix I assume that the user have the SDK installed already. Is it just a matter of setting up some paths so Clang can find them?

Perhaps the most recent version is looking in a different default directory?

The problem was with version 3.7. The earlier versions seems to use MinGW headers and later versions use Visual Studio headers. Version 3.7 do not detect neither by default (I don't really know why).

**Comment by jacob-carlborg** [comment]
Is this both 32 and 64bit ?

**Comment by ciechowoj** [comment]
Yes.

18.08.2016 22:07 "jacob-carlborg" <notifications@github.com> napisał(a):

File: clang/Cursor.d [hunk 1]
```diff
@@ -281,6 +281,11 @@ struct Cursor
         foreach (cursor; unit.cursor.children)
             result[cursor.spelling] = cursor.kind;
 
+        auto version_ = clangVersion();
+
+        if (version_.major == 3 && version_.minor == 7)
+            result["__int64"] = CXCursorKind.CXCursor_MacroDefinition;
```
Comment (jacob-carlborg) [inline clang/Cursor.d]: What is this for?

File: README.markdown [hunk 2]
```diff
@@ -36,6 +38,28 @@ The source code is available under the [Boost Software License 1.0](http://www.b
...
+1. Install all requirements, see [above](#requirements)
+2. Clone the repository by running:
+
+		$ git clone git://github.com/jacob-carlborg/dstep.git
+
+3. run `dub build --arch=x86_64 --build=release`
+
+#### Remarks
+
+Currently only 64-bit release version is supported. Remeber to install 64-bit version of LLVM to the default destination (`Program Files`) and to add its binaries to PATH enviromental variable.
```
Comment (jacob-carlborg) [inline README.markdown]: The documentation need to be updated now when Windows 32bit is supported.

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          Invoke-WebRequest http://llvm.org/releases/3.8.1/LLVM-3.8.1-win32.exe -OutFile c:\LLVM.exe;
+          7z x c:\LLVM.exe -oc:\PROGRA~2\LLVM > $null;
+        }
+
+  - set PATH=c:\dub;%PATH%
+  - dub --version
+
+before_build:
+  - set PATH=c:\dmd2\windows\bin;%PATH%
+  - ps: $env:compilersetup = "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall";
```
Comment (jacob-carlborg) [inline appveyor.yml]: Is the virtual machine always running Windows 64bit and we're cross-compiling to 32bit?

File: README.markdown [hunk 4]
```diff
@@ -21,13 +21,15 @@ The source code is available under the [Boost Software License 1.0](http://www.b
 
 ## Building
 
-### Requirements
+### Linux
```
Comment (jacob-carlborg) [inline README.markdown]: Linux -> Posix.

**Comment by jacob-carlborg** [comment]
It seems like the appveyor.yml file contains some configuration for running the tests. What if one wants to run the tests locally?

**Comment by jacob-carlborg** [comment]
The discussion about moving some content of `appveyor.yml` to script files. You mentioned that it needed to set environment variables that are required in later steps. What is the advantage of using all these different sections, like "install", "build_script" and so on. It would be possible to put everything in one script file and invoke that from one of the sections, right?

At least I don't see any indication of these sections in the UI for a given build.

File: clang/Cursor.d (hunk 1 above)
Comment (ciechowoj) [inline clang/Cursor.d]: Version 3.7 doesn't recognize __int64 and it is used in MinGW. It adds a macrodefinition for __int64 as long long.

File: appveyor.yml (hunk 3 above)
Comment (ciechowoj) [inline appveyor.yml]: Yes.

**Comment by ciechowoj** [comment]
> It seems like the appveyor.yml file contains some configuration for running the tests. What if one wants to run the tests locally?

Configuration in appveyor.yml it is mostly for CI. `dub --config=test` should run fine most of the time.

> The discussion about moving some content of appveyor.yml to script files. You mentioned that it needed to set environment variables that are required in later steps. What is the advantage of using all these different sections, like "install", "build_script" and so on. It would be possible to put everything in one script file and invoke that from one of the sections, right?

I'll check what can be done about it. 

This PR focuses on making it work for CI. I suppose some things can be improved to increase the probability that the end user builds dstep without problems. But it deserves its own PR.

File: clang/Cursor.d (hunk 1 above)
Comment (jacob-carlborg) [inline clang/Cursor.d]: I see. Is it only that version that is using MinGW?

Comment (ciechowoj) [inline clang/Cursor.d]: No the versions 3.4, 3.5, 3.6 are using MinGW too, but they work fine (which is a bit strange...).

**Comment by jacob-carlborg** [comment]
> This PR focuses on making it work for CI. I suppose some things can be improved to increase the probability that the end user builds dstep without problems. But it deserves its own PR.

The main reason why I would like to move it to separate file(s) is because it's then possible to take advantage of your editor: syntax highlighting, snippets, syntax validation and so on.

**Comment by ciechowoj** [comment]
> The main reason why I would like to move it to separate file(s) is because it's then possible to take advantage of your editor: syntax highlighting, snippets, syntax validation and so on.

I see.

File: README.markdown (hunk 2 above)
Comment (ciechowoj) [inline README.markdown]: Fixed.

File: README.markdown (hunk 4 above)
Comment (ciechowoj) [inline README.markdown]: Fixed.

**Comment by ciechowoj** [comment]
It took more time than it was worth,  but the script is in the separate file...

**Comment by ciechowoj** [comment]
You need to go to AppVeyor and cancel some test because their got stuck.

**Comment by jacob-carlborg** [comment]
Looking good, I just need the an update of the changelog to merge.

**Comment by ciechowoj** [comment]
Changelog updated.

**Comment by jacob-carlborg** [comment]
👌

**Comment by ciechowoj** [comment]
Thanks.

**Comment by jacob-carlborg** [comment]
BTW, have you tried using the OMF format? Since libclang is a DLL there's no problem, one just needs an import library in the correct format. This can be generated with `implib`, available here: http://www.digitalmars.com/download/freecompiler.html.

**Comment by ciechowoj** [comment]
No, I haven't tried. I'll check it out when I have some free time.

**Pull request merged**